### PR TITLE
Implement read resource endpoint

### LINKED ISSUE #223: Implement read resource endpoint
From the design document:

This method returns the details of the resource with the specified resource identifier:

```
GET /api/resource/KX3A-39WE HTTP/1.1
Authorization: Bearer MHg3OUZEQkZBMjcx
```

A successful response is indicated by a 200 OK header, and returns the details of the resource as a JSON document in the body.  The child_resource_ids parameter is a custom property (not part of the UMA specification) that lists the identifiers of resources that are child resources of the specified resource.

```
HTTP/1.1 200 OK
Content-Type: application/json
...
{  
   "_id":"KX3A-39WE",
   "resource_scopes":[  
      "read-public",
      "post-updates",
      "read-private",
      "http://www.example.com/scopes/all"
   ],
   "child_resource_ids":[
      "JK31-4927”
    ]
   "icon_uri":"http://www.example.com/icons/sharesocial.png",
   "name":"Tweedl Social Service",
   "type":"http://www.example.com/rsrcs/socialstream/140-compatible"
   "parent_resource_id":"KX30-1234"
}
```

### PR DESCRIPTION
**Pull request by sbryzak**
Implementation for read resource endpoint, removes unnecessary description fields, adds first resource type (`openshift.io/resource/area`)

fixes #223

### CODE REVIEW

File: controller/resource.go [hunk 1]
```diff
@@ -34,7 +34,33 @@ func (c *ResourceController) Delete(ctx *app.DeleteResourceContext) error {
 
 // Read runs the read action.
 func (c *ResourceController) Read(ctx *app.ReadResourceContext) error {
-	return ctx.MethodNotAllowed()
+
+	if !token.IsServiceAccount(ctx) {
+		log.Error(ctx, map[string]interface{}{}, "Unable to register resource. Not a service account")
```
Comment (alexeykazakov) [inline controller/resource.go]: Unable to **read** resource...

File: authorization/resource/resource_type.go [hunk 2]
```diff
@@ -78,36 +76,13 @@ func (m *GormResourceTypeRepository) Load(ctx context.Context, id uuid.UUID) (*R
...
-// LookupOrCreate looks up the ResourceType record with the specified name.  If there is no such record, then
-// a new ResourceType will be created with the specified name and returned.
-func (m *GormResourceTypeRepository) LookupOrCreate(ctx context.Context, name string) (*ResourceType, error) {
+// Lookup looks up the ResourceType record with the specified name.  If there is no such record, then
+// a gorm.ErrRecordNotFound error will be returned.
+func (m *GormResourceTypeRepository) Lookup(ctx context.Context, name string) (*ResourceType, error) {
 	defer goa.MeasureSince([]string{"goa", "db", "resource_type", "lookupOrCreate"}, time.Now())
 
 	var native ResourceType
 	err := m.db.Table(m.TableName()).Where("name = ?", name).First(&native).Error
```
[the remark this answers is not in the document]
Comment (sbryzak) [reply authorization/resource/resource_type.go]: Unique index has been added for this column.

File: controller/resource.go (hunk 1 above)
Comment (sbryzak) [reply controller/resource.go]: Updated, thanks.

File: authorization/resource/resource_blackbox_test.go [hunk 3]
```diff
@@ -77,13 +77,11 @@ func (s *resourceBlackBoxTest) TestExistsResource() {
...
 	resource := createAndLoadResource(s)
 
-	resource.Description = "foo"
 	err := s.repo.Save(s.Ctx, resource)
 	require.Nil(s.T(), err, "Could not update resource")
 
-	updatedResource, err := s.repo.Load(s.Ctx, resource.ResourceID)
+	_, err = s.repo.Load(s.Ctx, resource.ResourceID)
 	require.Nil(s.T(), err, "Could not load resource")
-	assert.Equal(s.T(), updatedResource.Description, "foo")
```
[the remark this answers is not in the document]
Comment (sbryzak) [reply authorization/resource/resource_blackbox_test.go]: I'll add a test for this as part of #224

**Comment by fabric8cd** [comment]
@sbryzak snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-240-6`

File: controller/resource_blackbox_test.go [hunk 4]
```diff
@@ -188,21 +198,19 @@ func (rest *TestResourceREST) TestRegisterResourceWithParentResourceSetCreated()
...
 		Name:             "My new resource",
 		ParentResourceID: nil,
 		ResourceScopes:   resourceScopes,
 		ResourceID:       &resourceID,
 		ResourceOwnerID:  resourceOwnerID.String(),
-		Type:             "Area",
+		Type:             "openshift.io/resource/area",
 	}
 
 	test.RegisterResourceNotFound(rest.T(), rest.service.Context, rest.service, rest.securedController, payload)
```
Comment (alexeykazakov) [inline controller/resource_blackbox_test.go]: I don't see any tests which check Read action?

File: authorization/resource/resource.go [hunk 5]
```diff
@@ -85,6 +87,11 @@ func (m *GormResourceRepository) Load(ctx context.Context, id string) (*Resource
 		return nil, errs.WithStack(errors.NewNotFoundError("resource", id))
 	}
 
+	err = m.db.Table(ResourceType{}.TableName()).Where("resource_type_id = ?", native.ResourceTypeID).Find(&native.ResourceType).Error
```
[the remark this answers is not in the document]
Comment (sbryzak) [reply authorization/resource/resource.go]: I spent most of Friday trying to get Preload to work.  We have the simplest possible use case, however the only error that gorm was capable of giving me was something along the lines of "ResourceType failed to preload" with zero clue as to *why*.  I've read through the gorm docs multiple times, searched stackoverflow and github issues, none of it was any help.  This is a workaround to compensate for preload's failure.

File: authorization/resource/resource_type_blackbox_test.go [hunk 6]
```diff
@@ -72,6 +72,16 @@ func (s *resourceTypeBlackBoxTest) TestExistsResourceType() {
 	})
 }
 
+func (s *resourceTypeBlackBoxTest) TestDefaultResourceTypesExist() {
+	t := s.T()
+
+	t.Run("resource type exists", func(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (sbryzak) [reply authorization/resource/resource_type_blackbox_test.go]: So just check a simple count of the total rows in that table?

File: controller/resource_blackbox_test.go (hunk 4 above)
Comment (sbryzak) [reply controller/resource_blackbox_test.go]: TestRegisterResourceWithResourceIDSetCreated() tests read, also a new test in #249 tests read also.

Comment (sbryzak) [reply controller/resource_blackbox_test.go]: (see https://github.com/fabric8-services/fabric8-auth/pull/249/files#diff-4b125eae9f559a68ad43b9d1ae45cd18R220)

Comment (alexeykazakov) [reply controller/resource_blackbox_test.go]: Please add a test(s) for Read in this PR too.

**Comment by fabric8cd** [comment]
@sbryzak snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-240-7`

File: controller/resource_blackbox_test.go (hunk 4 above)
Comment (sbryzak) [reply controller/resource_blackbox_test.go]: Done.

**Comment by fabric8cd** [comment]
@sbryzak snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-240-8`

File: authorization/resource/resource_type_blackbox_test.go [hunk 7]
```diff
@@ -27,79 +26,19 @@ func (s *resourceTypeBlackBoxTest) SetupTest() {
...
-	require.Nil(s.T(), err, "Could not create resource type")
-
-	createdResourceType, err := s.repo.Load(s.Ctx, resourceType.ResourceTypeID)
-	require.Nil(s.T(), err, "Could not load resource type")
-	require.Equal(s.T(), resourceType.Name, createdResourceType.Name)
-	require.Equal(s.T(), resourceType.ResourceTypeID, createdResourceType.ResourceTypeID)
-	require.Equal(s.T(), resourceType.Description, createdResourceType.Description)
-
-	return createdResourceType
+		fmt.Println("### number of resource types: ", len(types))
```
Comment (alexeykazakov) [inline authorization/resource/resource_type_blackbox_test.go]: Remove Println

Comment (sbryzak) [reply authorization/resource/resource_type_blackbox_test.go]: Done

**Comment by sbryzak** [comment]
I've added to TestFailRegisterResourceNonServiceAccount() a check, that attempts to read a resource with a non-service account (since the resource must first be registered anyway).

**Comment by fabric8cd** [comment]
@sbryzak snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-240-9`

**Comment by fabric8cd** [comment]
@sbryzak snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-240-10`

**Pull request merged**